tools/tplg_parser/tplg_parser.c

```diff
@@ -1128,6 +1128,11 @@ int load_widget(void *dev, int dev_type, struct comp_info *temp_comp_list,
 	if (ret != 1)
 		return -EINVAL;
 
+	if (!temp_comp_list) {
+		fprintf(stderr, "load_widget: temp_comp_list argument NULL\n");
+		return -EINVAL;
```

Comment:
@ymdatta we should move this check before we allocate memory for widget so we dont have to free it in case temp_comp_list is NULL

Reply:
Updated the patch with the suggestion.